src/elements/common/logger/Logger.js

```diff
@@ -0,0 +1,177 @@
+// @flow
+import * as React from 'react';
+import noop from 'lodash/noop';
+import { EVENT_JS_READY } from './constants';
+import { METRIC_TYPE_PREVIEW, METRIC_TYPE_ELEMENTS_LOAD_METRIC } from '../../../constants';
+
+type ElementsMetric = {
+    component: ElementOrigin,
+    name: string,
+    sessionId: string,
+    timestamp: string,
+    type: MetricType,
+} & ElementsLoadMetricData;
+
+type Props = {
+    fileId?: string,
+    onMetric: (data: Object) => void,
+    children: React.ChildrenArray<React.Element<any>>,
+    source: ElementOrigin,
+    startMarkName?: string,
+};
+
+/**
+ * Converts a character, using it as a seed, to a random integer and returns it as a string.
+ *
+ * @param {string} c - A single character
+ * @returns {string} The random character
+ */
+function generateRandom(c: string): string {
```

Comment:
These functions should be inside their own utilities.
Also https://github.com/kelektiv/node-uuid

Reply:
Using uuid